indexer/src/block_listener.rs

```diff
@@ -107,23 +111,46 @@ impl BlockListener {
                 info!(target: INDEXER, "No candidate data found in the streamer message");
                 continue;
             }
-            info!(target: INDEXER, "Found {} candidate(s)", candidates_data.len());
 
-            let results = join_all(candidates_data.into_iter().map(|receipt| receipt_sender.send(receipt))).await;
+            {
+                let candidates_len = candidates_data.len();
+                info!(target: INDEXER, "Found {} candidate(s)", candidates_len);
+                if let Some(event_listener) = &listener {
+                    event_listener.num_candidates.inc_by(candidates_len as f64);
+                }
+            }
+
+            // TODOL try_send instead checking for capacity. Not to stall
+            let results = join_all(
+                candidates_data
+                    .into_iter()
+                    .map(|receipt| candidates_sender.send(receipt)),
+            )
+            .await;
             results.into_iter().collect::<Result<_, _>>()?;
         }
 
         Ok(())
     }
 
-    pub(crate) async fn start(self) -> Result<()> {
-        let sender = self.receipt_sender.clone();
-        tokio::select! {
-            result = self.process_stream() => result,
-            _ = sender.closed() => {
-                Ok(())
-            },
-        }
+    /// Filters indexer stream and returns receiving channel.
+    pub(crate) fn run(
+        &self,
+        indexer_stream: Receiver<StreamerMessage>,
+    ) -> (JoinHandle<Result<()>>, Receiver<CandidateData>) {
+        let (candidates_sender, candidates_receiver) = mpsc::channel(100);
```

Comment:
It's enough, but while we're at this I don't think there's a good reason not to increase this especially since it's only created once and the NEAR transaction rate is high - let's say there's a period of very high activity. I'd say 1000 or even more is not an issue here.

Reply:
Don't mind that